pennylane/devices/tests/test_tracker.py

```diff
@@ -26,56 +26,25 @@ def test_tracker_initialization(self, device):
 
         dev = device(1)
 
-        if not dev.capabilities().get("supports_tracker", False):
+        if isinstance(dev, qml.Device) and not dev.capabilities().get("supports_tracker", False):
             pytest.skip("Device does not support a tracker")
 
         assert isinstance(dev.tracker, qml.Tracker)
 
-    def test_tracker_updated_in_execution_mode(self, device, mocker):
+    def test_tracker_updated_in_execution_mode(self, device):
         """Tests that device update and records during tracking mode"""
 
         dev = device(1)
 
-        if not dev.capabilities().get("supports_tracker", False):
+        if isinstance(dev, qml.Device) and not dev.capabilities().get("supports_tracker", False):
             pytest.skip("Device does not support a tracker")
 
         @qml.qnode(dev, diff_method="parameter-shift")
         def circ():
             return qml.expval(qml.PauliX(wires=[0]))
 
-        spy_update = mocker.spy(dev.tracker, "update")
-        spy_record = mocker.spy(dev.tracker, "record")
-
         dev.tracker.active = False
-        circ()
-
-        assert spy_update.call_count == 0
-        assert spy_record.call_count == 0
-
-        dev.tracker.active = True
-        circ()
-
-        assert spy_update.call_count == 1
-        assert spy_record.call_count == 1
-
-    def test_tracker_batch_execute(self, device, mocker):
```

Comment:
Is this test case not relevant anymore?

Reply:
So technically any execution does use the batch execution pathway.  As you can see on line 49, I am checking that we have one batch.